Expanding on definition of button element (#13834)
mdn/content: files/en-us/web/html/element/button/index.md

---

> {{HTMLRef}}

Author: I think the empty line is needed.

```suggestion
{{HTMLRef}}

```

---

{{HTMLRef}}

>   

Reviewer: A detail, but you have added two extra spaces on that line. Elle we are good to merge this.
```suggestion

```

Author: Oh no! Good catch. The GitHub UI was doing this every time I hit enter and I didn't notice these.

{{HTMLRef}}

> The **`<button>`** [HTML](/en-US/docs/Web/HTML) is an interactive element that performs a programmable action, such as submitting a [form](/en-US/docs/Learn/Forms) or opening a dialog, when activated by a user, whether by mouse click keyboard key, finger touch, voice command, or other assistive technology.

Reviewer: I agree about the circularity of the original definition. I find the new definition maybe a bit too hard for beginners (the most likely audience to need this page). What about the slight rephrase?
```suggestion
The **`<button>`** [HTML](/en-US/docs/Web/HTML) is an interactive element activated by a user, whether by mouse click keyboard key, finger touch, voice command, or other assistive technology. It then performs a programmable action, such as submitting a [form](/en-US/docs/Learn/Forms) or opening a dialog.
```

Author: @teoli2003 I like your idea of breaking it up into two sentences. It was feeling a little long.

Taking your suggestion a little further (and including the fixes I committed in b24d708), what do you think about this?

> The <button> HTML element is an interactive element activated by a user, whether by mouse click, keyboard key, finger touch, voice command, or other assistive technology. Once activated, it then performs a programmable action, such as submitting a form or opening a dialog.

I think the "... user, whether..." part might also benefit from a rephrase. The comma has a pause I'm not sure about. Here's my idea for that alongside the previous considerations:

> The <button> HTML element is an interactive element activated by a user with a mouse, keyboard, finger, voice command, or other assistive technology. Once activated, it then performs a programmable action, such as submitting a form or opening a dialog.

Reviewer: I like the second one the best. Can you add a commit with it?

Author: Done in 10b6ec6!
